### PR TITLE
Issue-3605. Fetch a customer

### PR DESCRIPTION
**Pull request by kidinov**
#3605 1 and 2

I've decided that with a coroutines-based approach it's ok to put tests in the same PR as well because it's way cleaner and shorter than the event bus.

* Description. An API call, caching, models, and DTOs for `/customers/{id}` endpoint

### CODE REVIEW

File: tests/api/src/test/java/APITesting_WCCustomer.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        Map<String, String> pathParams = new HashMap<>();
+        pathParams.put("json", "true");
+        pathParams.put("locale", "en_US");
+        pathParams.put("status", "any");
+
+        RequestSpecBuilder requestBuilder = new RequestSpecBuilder()
+                .setBaseUri("https://public-api.wordpress.com")
+                .setBasePath("rest/v1.1/jetpack-blogs/173063404/rest-api/")
+                .addQueryParams(pathParams)
+                .setAuth(oauth2(System.getenv("API_TEST_OAUTH_KEY")));
```
Comment (kidinov) [inline tests/api/src/test/java/APITesting_WCCustomer.java:28]: Can I have this key? If not, then maybe at least a JSON that is returned from there?

Comment (0nko) [reply tests/api/src/test/java/APITesting_WCCustomer.java:28]: You can get a sample from https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-customers. Would that work?

**Review by 0nko — COMMENTED** [review]
The code looks good! 👍 

Would you mind just adding a simple test functionality to the example app. It can be really simple, just something that prints out fetched data on the screen.

File: tests/api/src/test/java/APITesting_WCCustomer.java (hunk 1 above)
Comment (kidinov) [reply tests/api/src/test/java/APITesting_WCCustomer.java:28]: If I got the test correctly, we go here to a specific `173063404` site which should have a deterministic response (so it's like an isolated site which only this test can touch). In order to access it, I need to have the API_TEST_OAUTH_KEY or if you have one you could execute this request, and then based on the response I can finish the test. Is this correct?

**Comment by kidinov** [comment]
@0nko The example app is number 7 in the list of subtasks I planned to do. I'm working on it currently. Is this ok if it comes as a separate PR?

File: tests/api/src/test/java/APITesting_WCCustomer.java (hunk 1 above)
Comment (0nko) [reply tests/api/src/test/java/APITesting_WCCustomer.java:28]: Sorry, trialees don't have access to secret keys. I ran the test and this is the result:

```
JSON path data doesn't match.
Expected: a collection with size <50>
  Actual: {id=1, date_created=2020-02-18T20:12:38, date_created_gmt=2020-02-18T20:12:38, date_modified=2020-03-09T17:08:17, date_modified_gmt=2020-03-09T17:08:17, email=brent.sessions@automattic.com, first_name=, last_name=, role=administrator, username=woomobileapitesting, billing={first_name=, last_name=, company=, address_1=, address_2=, city=, postcode=, country=, state=, email=, phone=}, shipping={first_name=, last_name=, company=, address_1=, address_2=, city=, postcode=, country=, state=}, is_paying_customer=false, avatar_url=https://secure.gravatar.com/avatar/fb409f96c4ccb689c8b1d42342dae3c7?s=96&d=mm&r=g, meta_data=[{id=17, key=wc_last_active, value=1590969600}, {id=21, key=community-events-location, value={ip=75.166.141.0}}, {id=22, key=jetpack_tracks_anon_id, value=jetpack:SP1kqDkOfCIfcU6oO32qnPZ2}, {id=23, key=jetpack_tracks_wpcom_id, value=126352425}, {id=2403, key=woocommerce_admin_activity_panel_inbox_last_read, value=1583773693554}], _links={self=[{href=https://woomobileapitesting.mystagingwebsite.com/wp-json/wc/v3/customers/1}], collection=[{href=https://woomobileapitesting.mystagingwebsite.com/wp-json/wc/v3/customers}]}}
```

**Comment by 0nko** [comment]
> The example app is number 7 in the list of subtasks I planned to do. I'm working on it currently. Is this ok if it comes as a separate PR?

Yup, that's fine 👍

File: tests/api/src/test/java/APITesting_WCCustomer.java (hunk 1 above)
Comment (kidinov) [reply tests/api/src/test/java/APITesting_WCCustomer.java:28]: Added some [assertions](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1902/commits/e4a5023d8cdca8329f60df98201b21277bbbd3c4)

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**